config/v1/types_kmsencryption.go

```diff
@@ -0,0 +1,50 @@
+package v1
+
+// KMSConfig defines the configuration for the KMS instance
+// that will be used with KMSEncryptionProvider encryption
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'AWS' ?  has(self.aws) : !has(self.aws)",message="aws config is required when kms provider type is AWS, and forbidden otherwise"
```

Comment:
Since we only want one configuration to be provided, maybe we could add something like:
```
[self.aws(,self.azure,self.gcp,...)].exists_one(c, c != "null")
```

Reply:
Disregard since we want to keep KMSProviderType around